### ISSUE TITLE
Touchable with href can still be clicked when disabled is true

### ISSUE THREAD

**Issue by rnike** [necolas/react-native-web#2298]
**The problem**

Hello, I recently found out the `props.disable` of Touchable will not prevent it from clicking if `props.href` is provided, although it prevents the onPress event to be fired, but still will be navigated to the distination.

**How to reproduce**

Simplified test case: https://codesandbox.io/s/cocky-smoke-pl5kog

Steps to reproduce:
1. render a `TouchableOpacity` or `TouchableHighlight` with `props.href="/"` and `props.disabled={true}`
2. click on the touchable

**Expected behavior**

Should not be navigated to "/" since the touchable is disabled

**Environment (include versions). Did this work in previous versions?**

* React Native for Web (version): TBC
* React (version): TBC
* Browser: TBC

**Comment by rnike** [comment]
Currently here's my workaround

```ts
import { TouchableOpacity, TouchableOpacityProps } from 'react-native';

export default (props: TouchableOpacityProps)=>{
  // @ts-expect-error
  return <TouchableOpacity {...props} pointerEvents={props.disabled ? 'none' : 'auto'} />
}
```

Not sure if this is intended or not, if this is a bug, I'm willing to make a pull request.

**Comment by necolas** [comment]
Fixed by 094bd0efc5ac1c566a81dec7ee825deae2c5a9a7
